src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -79,6 +80,7 @@
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
+import javax.annotation.CheckForNull;
```

Comment:
I thought edu.umd.cs.findbugs.annotations was the deprecated?

Reply:
`javax.annotation.CheckForNull` is the JSR-305 variant which has questionable licensing.
the `edu.umd.cs.findbugs` ones are not deprecated (with a suitably recent version of spotbugs)

Reply:
Actually in this repo `edu.umd.cs.findbugs.annotations.CheckForNull` etc. _are_ still deprecated, but only because it is still compatible with an old version of Jenkins. That will change. In the meantime we use the proper imports, e.g. https://github.com/jenkinsci/jenkins-test-harness/blob/711534844e8a24c857c45165600996391972b0af/src/main/java/org/jvnet/hudson/test/JenkinsRule.java#L61